src/clientlib/cl_subscription_manager.c

```diff
@@ -900,17 +904,25 @@ cl_sm_rpc_process(cl_sm_ctx_t *sm_ctx, cl_sm_conn_ctx_t *conn, Sr__Msg *msg)
         goto cleanup;
     }
 
+    if (NULL != subscription->private_ctx) {
+        memcpy(session, subscription->private_ctx, sizeof(sr_session_ctx_t));
+    }
```

Comment:
I am totally confused about this `if`, what is going on here? Isn't it just some leftover from your testing?

Reply:
Not exactly. Memcpy fails, when we pass null pointer. _Sysrepo_ tests crash/not pass, when there is no check if private_ctx is null. Maybe more elegant is use just
`if (subscription->private_ctx) {` ?

Reply:
But I did not find in any other changes you made that you would set the `private_ctx` anywhere. Also, if you do, then you should not because it is meant to be used by the user, not internally.

Reply:
the `netconf_id` value is passed by np2srv_sr_rpc_send according to https://github.com/CESNET/Netopeer2/pull/376. User must pass session as private_ctx to `sr_rpc_subscribe`.

Reply:
In the *netopeer2-server* pull request it seems fine but I am no closer to understanding of these changes than I was before. Maybe it is my fault but my general idea would be that you set the NETCONF SID to a *sysrepo* session you are using to send the RPC (for example). It will go through the whole *sysrepo* and appear in the callback session *sysrepo* provides for the RPC subscriber. There it can be read from this session, that is what I think you want. All your changes fit into this idea, I think, except for this one `if` clause.

Why would then the `private_ctx` of `sr_rpc_subscribe` be needed to be set to anything by the user?